### PR TITLE
Remove non-API docs and tutorials (Cherry-pick of #11352)

### PR DESCRIPTION
**Pull request by Eric-Arellano**
See https://github.com/Qiskit/qiskit/pull/11352.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -18,9 +18,6 @@
```
Comment (Eric-Arellano) [inline docs/conf.py:18]: FYI this file had merge conflicts. We had changed config that is now deleted.

File: docs/api_redirects.txt [hunk 2]
```diff
@@ -1,286 +0,0 @@
-qiskit.algorithms.AlgorithmError algorithms
```
Comment (Eric-Arellano) [inline docs/api_redirects.txt:1]: FYI this file had merge conflicts. We had removed entries in 0.46, but now the whole file is deleted.

File: tox.ini [hunk 3]
```diff
@@ -73,12 +73,10 @@ setenv =
   {[testenv]setenv}
```
Comment (Eric-Arellano) [inline tox.ini:80]: This had merge conflicts. I messed it up in the first and second commits, but I think it's now right.

**Review by ElePT — APPROVED** [review]
I have crossed checked the changes with #11352 and it looks like everything is in place. The only difference is a change in a file that doesn't exist in this branch (`releasenotes/notes/platform-support-f7f693aaf5dec044.yaml`). If the docs build worked in main I assume it will be fine here too.

**Pull request merged**